### PR TITLE
Fetch instrumentation and Network failure method

### PR DESCRIPTION
**Pull request by ndesai-newrelic**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -120,6 +120,45 @@ window.XMLHttpRequest.prototype.open = function (
   return originalXhrOpen.apply(this, arguments as any);
 };
 
+const oldFetch = window.fetch;
+
+window.fetch = function fetch() {
```
Comment (metal-messiah) [inline src/index.ts:125]: Do you guys add/handle DT headers elsewhere?

File: src/index.ts [hunk 2]
```diff
@@ -175,3 +214,20 @@ window.XMLHttpRequest.prototype.send = function (
...
+
+function handleFetchSuccess(response: Response, method: string, url: string, startTime: number) {
+  NewRelicCapacitorPlugin.noticeHttpTransaction({
+    url:url,
+    method:method,
+    status:response.status,
+    startTime:startTime,
+    endTime:Date.now(),
+    bytesSent:0,
+    bytesReceived:0,
```
Comment (metal-messiah) [inline src/index.ts:226]: Do y'all not capture request sizes?

Comment (ndesai-newrelic) [reply src/index.ts:226]: how can I read  readablestream from response? I will add bytesSent from request-body

File: src/index.ts (hunk 1 above)
Comment (ndesai-newrelic) [reply src/index.ts:120]: i added it now.

File: src/index.ts (hunk 2 above)
Comment (ndesai-newrelic) [reply src/index.ts]: @metal-messiah i added that also

**Review by ywang-nr — APPROVED** [review]

**Pull request merged**